### ISSUE TITLE
extension frames

### ISSUE THREAD

**Issue by icing** [nghttp2/nghttp2#476]
Question: I see work on extension frames handling in callbacks. This is great! Looking forward to try that with CACHE_DIGEST... 

Is there any way an application can access such an "unknown" frame when using 1.6.0?

**Comment by tatsuhiro-t** [comment]
Good to know that you noticed upcoming new feature.
Before the PR is merged, there is no way to get "unknown" frame by application.
Those unknown frames to libnghttp2 are respectfully ignored by libnghttp2.

I'm also planning to experiment CACHE_DIGEST to make sure that our extension frame API really works.  The time frame to merge is later next month in earliest.  I'm not sure I have time in the next month, so there is no promise.

**Comment by icing** [comment]
Ok, thanks for the quick reply. Sure am willing to test, once you are able to put some time into it.

**Comment by tatsuhiro-t** [comment]
non-critical frame support was landed in master.  See nghttp2_unpack_extension_callback, nghttp2_on_extension_chunk_recv_callback, and nghttp2_option_set_user_recv_extension_type.

**Comment by tatsuhiro-t** [comment]
To send extension frame, use https://nghttp2.org/documentation/nghttp2_submit_extension.html#nghttp2-submit-extension

**Comment by icing** [comment]
Ah, thanks! That will work nicely.
